File: sassfnb/src/main/java/com/sassfnb/application/service/PublicMenuService.java
package com.sassfnb.application.service;

import com.sassfnb.adapters.rest.dto.publicmenu.PublicMenuDtos.PublicMenuTreeResponse;

import java.time.Instant;
import java.util.UUID;

public interface PublicMenuService {
    PublicMenuTreeResponse tree(UUID orderId, UUID outletId, Instant at);
}

File: sassfnb-fe/src/api/services/publicMenu.service.ts
// src/api/services/publicMenu.service.ts
import { publicHttp } from "../publicClient";
import type { PublicMenuTreeResponse } from "../../types/publicMenu";

export const publicMenuService = {
  // ✅ GET /api/v1/public/menu/tree?orderId=...&outletId=...&at=...
  tree: async (params: { orderId?: string; outletId?: string; at?: string }) => {
    const res = await publicHttp.get<PublicMenuTreeResponse>("/public/menu/tree", {
      params,
    });
    return res.data;
  },
};
